src/amo/components/Addon/index.js

```diff
@@ -69,8 +71,19 @@ export class AddonBase extends React.Component {
     getClientCompatibility: _getClientCompatibility,
   }
 
+  componentWillMount() {
+    const { addon, dispatch } = this.props;
+
+    dispatch(setCurrentView({ addonType: addon.type }));
```

Comment:
I don't think this is the right place to dispatch but it's ok for now. I think we need to move to a saga first. After that we can do something like this in the saga:
```js
export function* fetchAddon() {
  // This is a mythical saga to load an add-on for the detail page.
  const api = yield select(getApi);
  const response = yield call(getAddon, { api, addonSlug });
  const addon = yield put(setAddon(response));
  // After the add-on is loaded, we can dispatch its type.
  yield put(setAddonType(addon.type));
}
```